docs: update errors page with more details (#2196)
nextauthjs/next-auth: www/docs/errors.md

#### OAUTH_CALLBACK_HANDLER_ERROR

This error will occur when there was an issue parsing the json request body, for example.

There should also be further details logged when this occurs, such as the error thrown, and the request body itself to aid in debugging.

---

### Signin / Callback

#### GET_AUTHORIZATION_URL_ERROR

This error can occur when we cannot get the OAuth v1 request token and generate the authorization URL.

Please double check your OAuth v1 provider settings, especially the OAuth token and OAuth token secret.

#### SIGNIN_OAUTH_ERROR

This error can occur in one of a few places, first during the redirect to the authorization URL of the provider. Next, in the signin flow while creating the PKCE code verifier. Finally, during the generation of the CSRF Token hash in internal state during signin.

Please check your OAuth provider settings and make sure your URLs and other options are correctly set on the provider side.

#### CALLBACK_OAUTH_ERROR

This can occur during handling of the callback if the `code_verifier` cookie was not found or an invalid state was returned from the OAuth provider.

#### SIGNIN_EMAIL_ERROR

> This error can occur if there was an error signing in a user via Email link, such as if the email token could not be generated or the verification request failed.

Reviewer: ```suggestion
This error can occur when a user tries to sign in via an email link; for example, if the email token could not be generated or the verification request failed.
```
_I think_ email should be in lowercase here and re-wrote the sentence for clarity 👍🏽

Author: Yeah nice improvement!
